index.js

```diff
@@ -1,10 +1,16 @@
 /*!
  * word-wrap <https://github.com/jonschlinkert/word-wrap>
  *
- * Copyright (c) 2014-2017, Jon Schlinkert.
+ * Copyright (c) 2014-2023, Jon Schlinkert.
  * Released under the MIT License.
  */
 
+function trimTabAndSpaces(str) {
+  const lines = str.split('\n');
```

Comment:
may be using method chaining eliminates the need to create an intermediate variable ??
return str.split('\n').map(line => line.trimEnd()).join('\n');

Reply:
yup! can be done. I once just need verification the approach or whether it is important or not

Reply:
utilizing native `String.prototype.trimEnd()` would require the node-[engine](https://github.com/jonschlinkert/word-wrap/blob/master/package.json#L28) to be at least on version 10.0.0.

Ref: [String.prototype.trimEnd() - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd)

This should probably considered a breaking-change which requires new major version release.
Hence I would recommend to utilize custom implementation for this patch.

Reply:
oh! I can write `trimEnd` myself if that's the case

Reply:
For the reason mentioned by @mscheid-sf, for now let's merge in the regex I suggested in my other comment. After that I'd be happy to take another PR that replaces the regex with `.trimEnd()` (Using `.trimEnd` would indeed qualify as a breaking change according to semver, and also it's generally a good practice to minimize potential for regressions when releasing patches).